File: Data Structures & Algorithms/max-water-container/submission-7.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        length=len(heights)-1
        i,j=0,length
        res=0
        while i<j:
            if heights[i]<=heights[j]:
                res=max(res,length*heights[i])
                i+=1
            else:
                res=max(res,length*heights[j])
                j-=1
            length-=1
        return res

File: Data Structures & Algorithms/trapping-rain-water/submission-7.java
class Solution {
    public int trap(int[] height) {
        int left=height[0],right=height[height.length-1],l=1,r=height.length-2,res=0;
        for(;l<=r;){
            if(left<=right){
                left=Math.max(left,height[l]);
                res+=left-height[l];
                l++;
            }
            else {
                right=Math.max(right,height[r]);
                res+=right-height[r];
                r--;
            }
        }
        return res;
    }
}
